### ISSUE TITLE
process refinemen: auto deploy from a different branch than "main"

### ISSUE THREAD

**Issue by mr-c** [common-workflow-language/user_guide#256]
To facilitate review by @mr-c and others, we should stop auto deploying from the `main` branch and create a `release` branch or similar.

This would enable me and others to batch up reviews for weekly or twice-monthly review + deployment cycles.

Especially for Outreachy, this will enable @swzCuroverse and @tetron to process contributions without needing to wait for me.

This is also an opportunity to be thoughtful about release notes, credits, and confirming that existing links aren't broken.

Once this is setup , I will adjust merge permissions and I'll probably unsubscribe from most notifications so that @tetron and @swzCuroverse can focus on Outreachy.

**Comment by tetron** [comment]
Having a separate release branch is a good idea.  Can there be a persistent "latest main" rendered somewhere as well?  I would also expect to have the permission to merge to release with the understanding that you want an opportunity to do additional review before it goes to live.

**Comment by mr-c** [comment]
The easy hack is a standing PR to the release branch from the dev branch, and then using the read the docs preview. I'm sure there would be other options with more work than that.

**Comment by kinow** [comment]
Out of the loop, had a quick read, and remembered that there are also GH actions with an approval process. It requires someone from a group of users with permission to approve the workflow to run. Just in case it sounds helpful to use for this new process :+1:

**Comment by swzCuroverse** [comment]
Although the release branch is a nice thing to introduce.  It still not solve my original ask... There are 3 issues.

(1) This still bottlenecks on you - @mr-c - and you have said yourself that you are very busy right now with your PhD work.  Whether it is in a release or just waiting in the main - it is still a bottleneck to getting the content out to people who want to use it. There is no reason in free and open source project with both a leadership team and technical team that there should be bottle neck on 1 person. 

(2) There is no release process that I can find.  For this to be a viable solution, you will need to document how you want this to be done. including testing for broken links, what goes into release notes and who reviews them.   Otherwise, it again just hangs around until @mr-c can find time in his very busy schedule to do this.  This again is not a good idea from the standpoint that we are bottle-necking on @mr-c 

(3) There are no rules for who is giving permission over who can do a release.  Peter used to - but you with no reason or warning removed him.  I would like you to speak to that , explain why or return him to his original permissions.

**Comment by swzCuroverse** [comment]
Several thoughts.

* It seems that you are confusing returning Peter to his previous role in the CWL repos (i.e. the same permissions he used to have and all the leadership team currently has (except for Michael). We need transparency in this. Who has what permissions and why and why were Peter's permissions changed. I respectfully ask again that Peter be giving the same permissions as the rest of the leadership team. 

As for your suggestion. I would like to ask a few questions and make some suggestions.

* Since we haven't used this release process before with User guide, I would suggest we put off the change until we have time to work out the kinks and make a change while Outreachy folks are trying to contribute.

* Are we having specific named release branches i.e. User Guide 1.2,.1.3,. Etc Or are we just having a single release branch? 

* What does the public facing User Guide point to? The release branch or do we have a latest repo or main repo that changes are pushed to once a release is deemed finished. If that is the case, who and how do we determine a release is finished and who then merges it with this latest repo? 

* I personally think of the User guide to be documentation. I don't understand why we would need release notes? I think we should check links but not sure why we can't do that with the process now.

* I am not sure why we need a release repo to recognize authors or people who work on the User Guide.  Currently, we have an author's list on the readme and it is empty. Editing that and perhaps putting what people have done would be a good first way to do this? 

* Who would in charge of writing release notes? 

* If @mr-c is the only one who can declare a release ro be latest, write release note, determine if a "release" is done, etc then we still have a bottleneck problem.

**Comment by mr-c** [comment]
I clarified a few points in my proposal. As for who else could help with the release process, I would be happy to consider other people who do not work for a commercial CWL implementation, to avoid the conflict of interest problems we had in the past.

**Comment by mr-c** [comment]
I've created the `release` branch and updated the deployment action to run from there only; I've upgraded @tetron access; he can accept and merge PRs to the `main` branch

**Comment by swzCuroverse** [comment]
> I clarified a few points in my proposal. As for who else could help with the release process, I would be happy to consider other people who do not work for a commercial CWL implementation, to avoid the conflict of interest problems we had in the past.

That is pretty ridiculous @mr-c . Did the leadership team agree to this. Peter has built a large chunk of CWL with Seven Bridges. The only issue in the past was not with the website and quickly resolved.

**Comment by swzCuroverse** [comment]
You do understand everyone with an implementation they are funding via grants or customer support has a similar "conflict of interest". I don't think it is breaking the rules of our community for you to change this without discussing it with both the technical team AND the leadership team AND SFC. Additionally, I believe at BOSC you told Peter you would return his permissions. I guess you changed your mind?

**Comment by swzCuroverse** [comment]
Additionally we still have the issue that what is released to the public is still hung up on you @mr-c is still a problem.

**Comment by mr-c** [comment]
I'm talking about the final review before publishing updates to the user guide.

Contributions are welcome from all regardless of affiliation, of course.

**Comment by swzCuroverse** [comment]
> I'm talking about the final review before publishing updates to the user guide.
> 
> Contributions are welcome from all regardless of affiliation, of course.

Yes, I know. But this means ultimately you are the final decision of what goes into the User Guide (and every other repo you are the only person to release) and that is then this is not  decision of the community, leadership team or technical but your decision. Having a single person decide what goes into a project is really against the idea that it  is a community standard. It then becomes a Michael standard

**Comment by swzCuroverse** [comment]
> I'm talking about the final review before publishing updates to the user guide.
> 
> Contributions are welcome from all regardless of affiliation, of course.

And you didn't address why a commerical implementation is any different than a grant funded one. You didn't address that if seems you made this decision without any leadership or community discussion, and we still have a bus problem

**Comment by swzCuroverse** [comment]
What would make more sense to me if anyone on the technical team could make a release candidate. There is a reasonable time for those in tech or leadership team to veto for larger discount. If there is no veto, then the candidate can go out. This way we don't have a bus problem, there is no slow down since you are busy with your PhD, any "conflicts" of interest can be addressed if they pop up, and we have more a community based release.
